File: TwoSum.cs
public class Solution
{
    //O(n)
    //O(n)
    public int[] TwoSum(int[] nums, int target)
    {
        if (nums.Length <= 1) return new int[] { };
        var dict = new Dictionary<int, int>();
        
        for (int i = 0; i < nums.Length; i++)
        {
            int result = target - nums[i];
            if (dict.ContainsKey(result)) return new int[] { i, dict[result] };
            dict[nums[i]] = i;
        }

        return new int[] { };
    }
}

File: Go/twoSum.go
package main

func twoSum(nums []int, target int) []int {
	if len(nums) <= 0 {
		return []int{}
	}

	var dict = make(map[int]int)

	for i := 0; i < len(nums); i++ {
		var diff = target - nums[i]

		if _, ok := dict[diff]; ok {
			return []int{dict[diff], i}
		}

		dict[nums[i]] = i
	}
	return []int{}
}
